.gitlab-ci.yml

```diff
@@ -0,0 +1,211 @@
+workflow:
+  rules:
+    - if: $CI_MERGE_REQUEST_IID
+    - if: $CI_COMMIT_TAG
+    - if: $CI_COMMIT_BRANCH == $CI_DEFAULT_BRANCH
+
+.job-template: &job-template
+  image: debian:10
+  before_script:
+    - ulimit -c unlimited
+    - echo "core.%e.%p.%t" > /proc/sys/kernel/core_pattern
+    - apt-get update && apt-get install -y git wget unzip docker.io
+    - .travis/before_install.sh
+    - docker build -t cpp_tango .travis/${OS_TYPE}
+    - >
+        docker run
+        --rm
+        --name cpp_tango
+        --user root
+        -e BINTRAY_API_KEY=${CI_BINTRAY_API_KEY}
+        -e BINTRAY_USER_NAME=tango-ci
+        -e COVERALLS_REPO_TOKEN=${COVERALLS_REPO_TOKEN}
+        -e SONAR_TOKEN=${SONAR_TOKEN}
+        -e TANGO_HOST=${TANGO_HOST}
+        -e TRAVIS_BRANCH=${TRAVIS_BRANCH}
+        -e TRAVIS_JOB_ID=${TRAVIS_JOB_ID}
+        -e DOCKER_HOST=tcp://$(getent hosts docker | awk '{ print $1 }'):2375
+        -v `pwd`:/home/tango/src
+        -v `pwd`/.sonar:/home/tango/.sonar
+        -v `pwd`/idl:/home/tango/idl
+        -v `pwd`/cppzmq:/home/tango/cppzmq
+        -v `pwd`/tango_admin:/home/tango/tango_admin
+        -v `pwd`/coveralls-cmake:/home/tango/coveralls-cmake
+        -v `pwd`/build-wrapper-linux-x86:/home/tango/build-wrapper-linux-x86
+        -v `pwd`/sonar-scanner:/home/tango/sonar-scanner
+        -dit
+        cpp_tango
+    - .travis/install_tango_idl.sh
+    - test ${STOCK_CPPZMQ} = "OFF" && .travis/install_cppzmq.sh || true
+  script:
+    - set -e
+    - .travis/run.sh
+    - .travis/install_tango.sh
+    - .travis/install_tango_admin.sh
+    - .travis/test.sh
+    - set +e
+    - test ${SONAR_SCANNER} = "ON" && .travis/sonar.sh || true
+  after_script:
+    - docker stop cpp_tango
+    - find build -name 'core.*' -exec gzip "{}" \;
+    - mkdir -p build/cpp_test_suite/test_results
+    - gzip -r build/cpp_test_suite/test_results
+  artifacts:
+    when: always
+    paths:
+      - build/cpp_test_suite/test_results
+      - build/cpp_test_suite/*/core.*
+
+abi-api-compliance-check:
+  variables:
+    <<: *variables
+    RUN_TESTS: "OFF"
+  rules:
+    # This job runs only for merge requests.
+    - if: $CI_MERGE_REQUEST_IID
+  image: debian:10
+  before_script:
+    - apt-get update && apt-get install -y git docker.io
+    - git fetch origin "$CI_MERGE_REQUEST_TARGET_BRANCH_NAME"
+    - .travis/before_install.sh
+    - docker build -t cpp_tango .travis/gcc-latest
+    - >
+        docker run
+        --rm
+        --name cpp_tango
+        --user root
+        -v `pwd`:/home/tango/src
+        -v `pwd`/idl:/home/tango/idl
+        -dit
+        cpp_tango
+    - .travis/install_tango_idl.sh
+  script:
+    - >
+        docker exec
+        -w /home/tango/src
+        -e CI_TARGET_BRANCH=${CI_MERGE_REQUEST_TARGET_BRANCH_NAME}
+        -e CMAKE_BUILD_PARALLEL_LEVEL=$(nproc)
+        cpp_tango
+        .travis/check-ABI-API-compliance.sh || true
+  after_script:
+    - docker stop cpp_tango
+    - gzip -r compat_reports
```

Comment:
If we don't gzip the compat reports, it might be possible to browse them more easily online?: https://gitlab.com/tango-controls/test_gh_import/cppTango/-/jobs/956206341/artifacts/browse/compat_reports/libtango/fe934f0ccf09980e679e1f87b0bd042573a31a9b_to_46b5052e3f4576e0f5274c5d6cbbba58bde6a897/
What do you think?

Reply:
Yes, compression is not needed here. I guess Gitlab could display file contents directly in the artifacts UI: https://gitlab.com/gitlab-org/gitlab/-/issues/211517

I just pushed a commit that removes compression. Let's wait and see how it looks like: https://gitlab.com/tango-controls/test_gh_import/cppTango/-/jobs/960823927

Reply:
We can indeed browse html reports directly in the gitlab UI. Thanks for this comment.

https://tango-controls.gitlab.io/-/test_gh_import/cppTango/-/jobs/960823927/artifacts/compat_reports/libtango/fe934f0ccf09980e679e1f87b0bd042573a31a9b_to_a0e273c48570abc5c7820c3ca3dc4b05c3594e29/compat_report.html